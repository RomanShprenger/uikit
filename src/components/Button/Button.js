import * as React from 'react';
import { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { noop } from 'lodash';
import cx from 'classnames';
import { resolveStyles } from 'utils';
import { withFocusedState } from 'hocs';

const stylePath = resolveStyles('Button/styles');
import(`../${stylePath}`).then();

const Button = (props) => {
  const {
    disabled,
    focused,
    onMouseEnter,
    onMouseLeave,
    onMouseDown,
    onMouseUp,
    onClick,
    component: Component,
    size,
    format,
    kind,
    classNamePrefix,
    className,
    onFocus,
    onBlur,
    type,
    ...other
  } = props;

  const [hovered, setHovered] = useState(false);
  const [pressed, setPressed] = useState(false);

  useEffect(() => {
    if (disabled) {
      setHovered(false);
      setPressed(false);
    }
  }, [disabled]);

  const handleMouseEnter = (e) => {
    if (disabled) {
      return false;
    }

    setHovered(true);

    return onMouseEnter(e);
  };

  const handleMouseLeave = (e) => {
    if (disabled) {
      return false;
    }

    setHovered(false);
    setPressed(false);

    return onMouseLeave(e);
  };

  const handleMouseDown = (e) => {
    if (disabled) {
      return false;
    }

    setPressed(true);

    if (!focused) {
      setTimeout(() => typeof e.focus === 'function' && e.focus(), 0);
    }

    return onMouseDown(e);
  };

  const handleMouseUp = (e) => {
    if (disabled) {
      return false;
    }

    setPressed(false);

    return onMouseUp(e);
  };

  const handleClick = (e) => {
    if (disabled) {
      return false;
    }

    return onClick(e);
  };

  const mergedClassName = cx(
    classNamePrefix,
    {
      [`${classNamePrefix}--${size}`]: size,
      [`${classNamePrefix}--${format}`]: format,
      [`${classNamePrefix}--${kind}`]: kind,
      [`${classNamePrefix}--disabled`]: disabled === true,
      [`${classNamePrefix}--active`]: focused,
      [`${classNamePrefix}--hover`]: hovered && !focused,
      [`${classNamePrefix}--hover-active`]: hovered && focused,
      [`${classNamePrefix}--press`]: pressed,
    },
    className,
  );

  return (
    <Component
      {...{
        ...other,
        disabled,
        className: mergedClassName,
        onClick: handleClick,
        onFocus,
        onBlur,
        type: Component === 'button' ? type : null,
        onMouseEnter: handleMouseEnter,
        onMouseLeave: handleMouseLeave,
        onMouseDown: handleMouseDown,
        onMouseUp: handleMouseUp,
      }}
    />
  );
};

Button.propTypes = {
  component: PropTypes.oneOfType([PropTypes.func, PropTypes.string]),
  kind: PropTypes.oneOf(['primary', 'secondary', 'outline', 'inside', 'inverse']),
  size: PropTypes.oneOf(['s', 'm', 'l']),
  format: PropTypes.oneOf(['icon', 'file']),
  type: PropTypes.string,
  disabled: PropTypes.bool,
  focused: PropTypes.bool,
  classNamePrefix: PropTypes.string,
  className: PropTypes.string,
  onFocus: PropTypes.func,
  onBlur: PropTypes.func,
  onMouseEnter: PropTypes.func,
  onMouseLeave: PropTypes.func,
  onMouseDown: PropTypes.func,
  onMouseUp: PropTypes.func,
  onClick: PropTypes.func,
};

Button.defaultProps = {
  component: 'button',
  kind: 'primary',
  size: 'm',
  format: undefined,
  type: 'button',
  disabled: false,
  classNamePrefix: 'yui-button',
  className: '',
  focused: false,
  onFocus: noop,
  onBlur: noop,
  onMouseEnter: noop,
  onMouseLeave: noop,
  onMouseDown: noop,
  onMouseUp: noop,
  onClick: noop,
};

export default withFocusedState(Button);
