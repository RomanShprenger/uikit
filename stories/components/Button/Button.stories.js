import React from 'react';
import * as storybook from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { withKnobs, boolean, text, number } from '@storybook/addon-knobs';
import { withActions } from '@storybook/addon-actions';

import { Button as RawButton, Icon } from 'components';

const Button = props => (<div style={{ marginRight: 5, display: 'inline-block' }}><RawButton {...props} /></div>);

export default storybook
  .storiesOf('components/Button', module)
  .addDecorator(withInfo({ propTables: [RawButton], propTablesExclude: [Button] }))
  .addDecorator(withActions('focus', 'blur', 'mouseenter', 'mouseleave', 'mousedown', 'mouseup', 'click'))
  .addDecorator(withKnobs)
  .add('Button', () => (
    <>
      <div style={{ padding: '20px' }}>
        <Button>Default</Button>
        <Button disabled={boolean('Disabled', true)}>Primary</Button>
        <Button size="s" kind="primary">Primary Small</Button>
        <Button size="l" kind="primary">Primary Large</Button>
        <Button size="s" kind="secondary" disabled={boolean('Disabled Small', false)}>Secondary Small</Button>
        <Button size="m" kind="secondary" disabled={boolean('Disabled Medium', false)}>Secondary Medium</Button>
        <Button size="l" kind="secondary" disabled={boolean('Disabled Large', false)}>Secondary Large</Button>
        <Button size="m" kind="outline">Outline</Button>
        <Button size="s" kind="outline" disabled={boolean('Disabled Small', false)}>Outline Small</Button>
        <Button size="l" format="icon" kind="primary">
          <Icon name={text('Icon name', 'arrow-left')} size={number('Size', 24)} />
        </Button>
        <Button size="s" format="icon" kind="secondary">
          <Icon name={text('Icon name', 'arrow-left')} size={number('Size', 24)} />
        </Button>
        <Button size="s" format="icon" kind="inside">
          <Icon name={text('Icon name', 'eye')} size={number('Size', 24)} />
        </Button>
        <Button size="m" format="file" kind="outline">
          <input type="file" accept=".png, .jpg, .gif" />
          Upload image
        </Button>
      </div>
      <div style={{ background: '#494952', padding: '20px' }}>
        <Button kind="inverse">Inverse</Button>
        <Button kind="inverse" disabled={boolean('Disabled', true)}>Inverse</Button>
      </div>
    </>
  ));
