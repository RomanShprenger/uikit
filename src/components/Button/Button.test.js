import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import Button from './Button';

Enzyme.configure({
  adapter: new Adapter(),
});

describe('Test: render of different variants of button', () => {
  it('Should render Button with default settings correctly', () => {
    const tree = renderer
      .create(<Button />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should render primary Button correctly', () => {
    const tree = renderer
      .create(<Button kind="primary" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should render secondary Button correctly', () => {
    const tree = renderer
      .create(<Button kind="secondary" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should render outline Button correctly', () => {
    const tree = renderer
      .create(<Button kind="outline" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should render inside Button correctly', () => {
    const tree = renderer
      .create(<Button kind="inside" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should render inverse Button correctly', () => {
    const tree = renderer
      .create(<Button kind="inverse" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should render small Button correctly', () => {
    const tree = renderer
      .create(<Button size="s" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should render medium Button correctly', () => {
    const tree = renderer
      .create(<Button size="m" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should render large Button correctly', () => {
    const tree = renderer
      .create(<Button size="l" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should render icon Button correctly', () => {
    const tree = renderer
      .create(<Button format="icon" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should render file Button correctly', () => {
    const tree = renderer
      .create(<Button format="file" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should render button type Button correctly', () => {
    const tree = renderer
      .create(<Button type="button" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should render submit Button correctly', () => {
    const tree = renderer
      .create(<Button type="submit" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should render reset Button correctly', () => {
    const tree = renderer
      .create(<Button type="reset" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should render disabled Button correctly', () => {
    const tree = renderer
      .create(<Button disabled />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should render focused Button correctly', () => {
    const tree = renderer
      .create(<Button focused />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should render focused and disabled Button correctly', () => {
    const tree = renderer
      .create(<Button focused />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should render Button with class prefix correctly', () => {
    const tree = renderer
      .create(<Button classNamePrefix="test" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('Test: markup of default button after user actions', () => {
  it('Should render Button after hover correctly', () => {
    const wrapper = mount(<Button />);
    wrapper.find('button').simulate('mouseenter');
    expect(toJson(wrapper, { mode: 'deep' })).toMatchSnapshot();
  });

  it('Should render Button after click correctly', () => {
    const wrapper = mount(<Button />);
    wrapper.find('button').simulate('mousedown');
    expect(toJson(wrapper, { mode: 'deep' })).toMatchSnapshot();
  });

  it('Should render Button after focus correctly', () => {
    const wrapper = mount(<Button />);
    wrapper.find('button').simulate('focus');
    expect(toJson(wrapper, { mode: 'deep' })).toMatchSnapshot();
  });

  it('Should render Button after focus and hover correctly', () => {
    const wrapper = mount(<Button />);
    wrapper.find('button').simulate('focus');
    wrapper.find('button').simulate('mouseenter');
    expect(toJson(wrapper, { mode: 'deep' })).toMatchSnapshot();
  });
});

describe('Test: markup of disabled button after user actions', () => {
  it('Should render Button after hover correctly', () => {
    const wrapper = mount(<Button disabled />);
    wrapper.find('button').simulate('mouseenter');
    expect(toJson(wrapper, { mode: 'deep' })).toMatchSnapshot();
  });

  it('Should render Button after click correctly', () => {
    const wrapper = mount(<Button disabled />);
    wrapper.find('button').simulate('mousedown');
    expect(toJson(wrapper, { mode: 'deep' })).toMatchSnapshot();
  });

  it('Should render Button after focus correctly', () => {
    const wrapper = mount(<Button disabled />);
    wrapper.find('button').simulate('focus');
    expect(toJson(wrapper, { mode: 'deep' })).toMatchSnapshot();
  });

  it('Should render Button after focus and hover correctly', () => {
    const wrapper = mount(<Button disabled />);
    wrapper.find('button').simulate('focus');
    wrapper.find('button').simulate('mouseenter');
    expect(toJson(wrapper, { mode: 'deep' })).toMatchSnapshot();
  });
});

describe('Test: markup of focused button after user actions', () => {
  it('Should render Button after hover correctly', () => {
    const wrapper = mount(<Button focused />);
    wrapper.find('button').simulate('mouseenter');
    expect(toJson(wrapper, { mode: 'deep' })).toMatchSnapshot();
  });

  it('Should render Button after click correctly', () => {
    const wrapper = mount(<Button focused />);
    wrapper.find('button').simulate('mousedown');
    expect(toJson(wrapper, { mode: 'deep' })).toMatchSnapshot();
  });

  it('Should render Button after focus correctly', () => {
    const wrapper = mount(<Button focused />);
    wrapper.find('button').simulate('focus');
    expect(toJson(wrapper, { mode: 'deep' })).toMatchSnapshot();
  });

  it('Should render Button after focus and hover correctly', () => {
    const wrapper = mount(<Button focused />);
    wrapper.find('button').simulate('focus');
    wrapper.find('button').simulate('mouseenter');
    expect(toJson(wrapper, { mode: 'deep' })).toMatchSnapshot();
  });
});
