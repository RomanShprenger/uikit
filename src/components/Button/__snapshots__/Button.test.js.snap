// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test: markup of default button after user actions Should render Button after click correctly 1`] = `
<button
  className="yui-button yui-button--m yui-button--primary yui-button--press"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  type="button"
/>
`;

exports[`Test: markup of default button after user actions Should render Button after focus and hover correctly 1`] = `
<button
  className="yui-button yui-button--m yui-button--primary yui-button--active yui-button--hover-active"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  type="button"
/>
`;

exports[`Test: markup of default button after user actions Should render Button after focus correctly 1`] = `
<button
  className="yui-button yui-button--m yui-button--primary yui-button--active"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  type="button"
/>
`;

exports[`Test: markup of default button after user actions Should render Button after hover correctly 1`] = `
<button
  className="yui-button yui-button--m yui-button--primary yui-button--hover"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  type="button"
/>
`;

exports[`Test: markup of disabled button after user actions Should render Button after click correctly 1`] = `
<button
  className="yui-button yui-button--m yui-button--primary yui-button--disabled"
  disabled={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  type="button"
/>
`;

exports[`Test: markup of disabled button after user actions Should render Button after focus and hover correctly 1`] = `
<button
  className="yui-button yui-button--m yui-button--primary yui-button--disabled"
  disabled={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  type="button"
/>
`;

exports[`Test: markup of disabled button after user actions Should render Button after focus correctly 1`] = `
<button
  className="yui-button yui-button--m yui-button--primary yui-button--disabled"
  disabled={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  type="button"
/>
`;

exports[`Test: markup of disabled button after user actions Should render Button after hover correctly 1`] = `
<button
  className="yui-button yui-button--m yui-button--primary yui-button--disabled"
  disabled={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  type="button"
/>
`;

exports[`Test: markup of focused button after user actions Should render Button after click correctly 1`] = `
<button
  className="yui-button yui-button--m yui-button--primary yui-button--press"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  type="button"
/>
`;

exports[`Test: markup of focused button after user actions Should render Button after focus and hover correctly 1`] = `
<button
  className="yui-button yui-button--m yui-button--primary yui-button--active yui-button--hover-active"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  type="button"
/>
`;

exports[`Test: markup of focused button after user actions Should render Button after focus correctly 1`] = `
<button
  className="yui-button yui-button--m yui-button--primary yui-button--active"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  type="button"
/>
`;

exports[`Test: markup of focused button after user actions Should render Button after hover correctly 1`] = `
<button
  className="yui-button yui-button--m yui-button--primary yui-button--hover"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  type="button"
/>
`;

exports[`Test: render of different variants of button Should render Button with class prefix correctly 1`] = `
<button
  className="test test--m test--primary"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  type="button"
/>
`;

exports[`Test: render of different variants of button Should render Button with default settings correctly 1`] = `
<button
  className="yui-button yui-button--m yui-button--primary"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  type="button"
/>
`;

exports[`Test: render of different variants of button Should render button type Button correctly 1`] = `
<button
  className="yui-button yui-button--m yui-button--primary"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  type="button"
/>
`;

exports[`Test: render of different variants of button Should render disabled Button correctly 1`] = `
<button
  className="yui-button yui-button--m yui-button--primary yui-button--disabled"
  disabled={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  type="button"
/>
`;

exports[`Test: render of different variants of button Should render file Button correctly 1`] = `
<button
  className="yui-button yui-button--m yui-button--file yui-button--primary"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  type="button"
/>
`;

exports[`Test: render of different variants of button Should render focused Button correctly 1`] = `
<button
  className="yui-button yui-button--m yui-button--primary"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  type="button"
/>
`;

exports[`Test: render of different variants of button Should render focused and disabled Button correctly 1`] = `
<button
  className="yui-button yui-button--m yui-button--primary"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  type="button"
/>
`;

exports[`Test: render of different variants of button Should render icon Button correctly 1`] = `
<button
  className="yui-button yui-button--m yui-button--icon yui-button--primary"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  type="button"
/>
`;

exports[`Test: render of different variants of button Should render inside Button correctly 1`] = `
<button
  className="yui-button yui-button--m yui-button--inside"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  type="button"
/>
`;

exports[`Test: render of different variants of button Should render inverse Button correctly 1`] = `
<button
  className="yui-button yui-button--m yui-button--inverse"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  type="button"
/>
`;

exports[`Test: render of different variants of button Should render large Button correctly 1`] = `
<button
  className="yui-button yui-button--l yui-button--primary"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  type="button"
/>
`;

exports[`Test: render of different variants of button Should render medium Button correctly 1`] = `
<button
  className="yui-button yui-button--m yui-button--primary"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  type="button"
/>
`;

exports[`Test: render of different variants of button Should render outline Button correctly 1`] = `
<button
  className="yui-button yui-button--m yui-button--outline"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  type="button"
/>
`;

exports[`Test: render of different variants of button Should render primary Button correctly 1`] = `
<button
  className="yui-button yui-button--m yui-button--primary"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  type="button"
/>
`;

exports[`Test: render of different variants of button Should render reset Button correctly 1`] = `
<button
  className="yui-button yui-button--m yui-button--primary"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  type="reset"
/>
`;

exports[`Test: render of different variants of button Should render secondary Button correctly 1`] = `
<button
  className="yui-button yui-button--m yui-button--secondary"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  type="button"
/>
`;

exports[`Test: render of different variants of button Should render small Button correctly 1`] = `
<button
  className="yui-button yui-button--s yui-button--primary"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  type="button"
/>
`;

exports[`Test: render of different variants of button Should render submit Button correctly 1`] = `
<button
  className="yui-button yui-button--m yui-button--primary"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  type="submit"
/>
`;
